python/ray/util/client/server/server.py

```diff
@@ -252,6 +254,32 @@ def Terminate(self, req, context=None):
     def GetObject(self, request, context=None):
         return self._get_object(request, "", context)
 
+    def _async_get_object(self, request, client_id: str, req_id: int, context):
+        if request.id not in self.object_refs[client_id]:
+            return ray_client_pb2.GetResponse(valid=False)
+        try:
+            assert request.asynchronous
+            assert context is not None
+            objectref = self.object_refs[client_id][request.id]
+            logger.debug("async get: %s" % objectref)
+            with disable_client_hook():
+                loop = asyncio.get_event_loop()
+
+                def f(result):
+                    serialized = dumps_from_server(result, client_id, self)
+                    resp = ray_client_pb2.DataResponse(
+                        get=ray_client_pb2.GetResponse(
+                            valid=True, data=serialized))
+                    resp.req_id = req_id
```

Comment:
can this fail/throw?

Reply:
In the normal use case it does not appear to be in a try except. That being said, I'll nest the `dumps_from_server` inside a `try except` block to ensure we don't have anything blowing up in the CoreWorker.